ddtrace/profiling/collector/threading.py

```diff
@@ -37,3 +41,25 @@ def _set_original(
     ):
         # type: (...) -> None
         threading.Lock = value  # type: ignore[misc]
+
+
+# Also patch threading.Thread so echion can track thread lifetimes
```

Comment:
Basically we need to do what echion's `monkey/threading.py` does, right?
https://github.com/sanchda/echion/blob/main/echion/monkey/threading.py

And the reason we're not able to invoke echion's Python code directly is that we're using its C++ APIs and we don't depend on its Python APIs.